tsdb/tsdbblockutil.go

```diff
@@ -61,30 +39,35 @@ func CreateBlock(samples []*MetricSample, dir string, mint, maxt int64, logger l
 	if chunkRange < 0 {
 		return "", ErrInvalidTimes
 	}
-	chunkDir := filepath.Join(dir, "chunks_tmp")
-	defer func() {
-		os.RemoveAll(chunkDir)
-	}()
-	head, err := CreateHead(samples, chunkRange, chunkDir, logger)
+
+	w := NewTSDBWriter(logger, dir)
+	err := w.initHead(chunkRange)
```

Comment:
Should the `chunkRange` be part of the `NewTSDBWriter` arguments and it calling the `initHead`? As `initHead` is an unexposed method and not a part of the interface, it cannot be used outside this package.